src/java.base/share/classes/java/lang/ref/Cleaner.java

```diff
@@ -215,6 +215,14 @@ public static Cleaner create(ThreadFactory threadFactory) {
      * Refer to the <a href="#compatible-cleaners">API Note</a> above for
      * cautions about the behavior of cleaning actions.
      *
+     * <p>The object being registered is kept strongly reachable (and therefore not eligible
+     * for cleaning) during the register() method.
+     *
+     * <p><a href="{@docRoot}/java.base/java/lang/ref/package-summary.html#MemoryConsistency">Memory consistency effects</a>:
+     * Actions in a thread prior to calling {@code Cleaner.register()}
+     * <a href="{@docRoot}/java.base/java/util/concurrent/package-summary.html#MemoryVisibility"><i>happen-before</i></a>
+     * the cleaning action is run by the Cleaner's thread.
```

Comment:
In passing, you could reduces the html refs to "package-summary.html#MemoryConsistency" and "package-summary.html#MemoryVisibility" as it's the same package/directory.

Reply:
MemoryVisibility is in java.util.concurrent 😉